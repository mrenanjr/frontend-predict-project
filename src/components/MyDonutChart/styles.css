.donutchart-track{
    fill: transparent;
    stroke: #DAE2E5;
    stroke-width: 26;
}
.donutchart-indicator {
    fill: transparent;
    stroke-width: 26;
    stroke-dasharray: 0 10000;
    transition: stroke-dasharray .3s ease;
}

.donutchart {
    border-radius: 50%;
    display: block;
}

.donutchart-text{
    fill: white;
}

.donutchart-text-val{
    font-size: 18px;
}

.donutchart-text-percent{
    font-size: 16px;
}

.donutchart-text-label{
    font-size: 16px;
}