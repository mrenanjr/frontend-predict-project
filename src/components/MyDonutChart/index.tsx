import React from 'react';

import './styles.css';

interface PropsInfo {
    value?: number,
    valuelabel: string,
    size: number,
    strokewidth: number
}

const MyDonutChart = (props: PropsInfo) => {
    const halfsize = (props.size * 0.5);
    const radius = halfsize - (props.strokewidth * 0.5);
    const circumference = 2 * Math.PI * radius;
    const strokeval = ((props.value! * circumference) / 100);
    const dashval = (strokeval + ' ' + circumference);

    const trackstyle = {strokeWidth: props.strokewidth};
    const indicatorstyle = {
        strokeWidth: props.strokewidth,
        strokeDasharray: dashval,
        stroke: props.value! > 55 ? 'red' : props.value! > 45 ? 'yellow' : '#009688' 
    }
    const rotateval = 'rotate(-90 '+halfsize+','+halfsize+')';

    return (
        <svg width={props.size} height={props.size} className="donutchart">
            <circle r={radius} cx={halfsize} cy={halfsize} transform={rotateval} style={trackstyle} className="donutchart-track"/>
            <circle r={radius} cx={halfsize} cy={halfsize} transform={rotateval} style={indicatorstyle} className="donutchart-indicator"/>
            <text className="donutchart-text" x={halfsize} y={halfsize} style={{textAnchor:'middle'}} >
                <tspan className="donutchart-text-val">{props.value}</tspan>
                <tspan className="donutchart-text-percent">%</tspan>
                <tspan className="donutchart-text-label" x={halfsize} y={halfsize+10}>{props.valuelabel}</tspan>
            </text>
        </svg>
    );
}

export default MyDonutChart;