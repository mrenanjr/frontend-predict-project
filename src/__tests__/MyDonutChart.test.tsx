import * as React from 'react';
import { shallow } from 'enzyme';
import MyDonutChart from '../components/MyDonutChart';

const TestDonutValues = {
    value: 70,
    valueLabel: 'Teste',
    size: 126,
    strokewidth: 26
}

test('MyDonutChart loads corretly', () => {
  const wrapper = shallow(<MyDonutChart 
        value={TestDonutValues.value}
        valuelabel={TestDonutValues.valueLabel}
        size={TestDonutValues.size}
        strokewidth={TestDonutValues.strokewidth}
    />);

  expect(wrapper).toMatchSnapshot();
});