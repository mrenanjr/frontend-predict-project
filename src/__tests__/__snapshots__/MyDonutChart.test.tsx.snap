// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDonutChart loads corretly 1`] = `
<svg
  className="donutchart"
  height={126}
  width={126}
>
  <circle
    className="donutchart-track"
    cx={63}
    cy={63}
    r={50}
    style={
      Object {
        "strokeWidth": 26,
      }
    }
    transform="rotate(-90 63,63)"
  />
  <circle
    className="donutchart-indicator"
    cx={63}
    cy={63}
    r={50}
    style={
      Object {
        "stroke": "red",
        "strokeDasharray": "219.91148575128554 314.1592653589793",
        "strokeWidth": 26,
      }
    }
    transform="rotate(-90 63,63)"
  />
  <text
    className="donutchart-text"
    style={
      Object {
        "textAnchor": "middle",
      }
    }
    x={63}
    y={63}
  >
    <tspan
      className="donutchart-text-val"
    >
      70
    </tspan>
    <tspan
      className="donutchart-text-percent"
    >
      %
    </tspan>
    <tspan
      className="donutchart-text-label"
      dy="0.5em"
      fill="#0F084B"
      x={63}
      y={73}
    >
      Teste
    </tspan>
  </text>
</svg>
`;
